src/App.tsx

```diff
@@ -1,26 +1,87 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+// App.tsx
 
-const USER_ID = 0;
+import React, { useEffect, useMemo, useState } from 'react';
+import { UserWarning } from './UserWarning';
+import { USER_ID, getTodos } from './api/todos';
+import { List } from './components/List';
+import { Error } from './components/Error';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [filter, setFilter] = useState(Filter.All);
+
+  useEffect(() => {
+    if (!USER_ID) {
+      return;
+    }
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  const addTodo = (newTodo: Todo) => {
+    setTodos(prevTodos => [...prevTodos, newTodo]);
+  };
+
+  const handleError = (message: string) => {
+    setErrorMessage(message);
+  };
+
+  const activeTodosCount = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  const filterTodos = useMemo(() => {
```

Comment:
recommend move it to another file, for example functions.tsx

Reply:
look